python_modules/dagster/dagster/utils/__init__.py

```diff
@@ -84,3 +84,42 @@ def merge_dicts(left, right):
     result = left.copy()
     result.update(right)
     return result
+
+
+class ReadOnlyDict(dict):
```

Comment:
ok, some would call these `frozendict` or `frozenlist`, we could vendor existing implementations, but w/e